src/Nethermind/Nethermind.Evm/ReadOnlyMemoryExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+
+namespace Nethermind.Evm
+{
+    public static class ReadOnlyMemoryExtensions
+    {
+        public static bool StartsWith(this ReadOnlyMemory<byte> inputData, byte startingByte)
+        {
+            return inputData.Span.Slice(0, 1)[0] == startingByte;
```

Comment:
Span has indexer [0], right? why to Slice it first?

Reply:
Oh, right thx :)